diff --git a/rngd_linux.c b/rngd_linux.c
index bcc0418..1e3c61c 100644
--- a/rngd_linux.c
+++ b/rngd_linux.c
@@ -160,10 +160,15 @@ int random_add_entropy(void *buf, size_t size)
 
 void random_sleep(void)
 {
+	static int ctr;
 	struct pollfd pfd = {
 		.fd = random_fd,
 		.events = POLLOUT,
 	};
 	malloc_trim(0);
-	poll(&pfd, 1, -1);
+	ctr++;
+	if (ctr > 1) {
+		poll(&pfd, 1, -1);
+		ctr = 0;
+	}
 }
